packages/expo-cli/lib/configure-plugin.js

```diff
@@ -24,4 +24,28 @@ module.exports = async (projectRoot) => {
     }
     throw e
   }
+
+  try {
+    const sourceMaps = '@bugsnag/source-maps'
+    const packageJsonPath = join(projectRoot, 'package.json')
+    const packageJson = JSON.parse(await promisify(readFile)(packageJsonPath))
+    packageJson.workspaces = packageJson.workspaces || {}
+    packageJson.workspaces.nohoist = packageJson.workspaces.nohoist || []
+    if (!packageJson.workspaces.nohoist.includes(plugin)) {
+      packageJson.workspaces.nohoist.push(plugin)
+    }
+    if (!packageJson.workspaces.nohoist.includes(sourceMaps)) {
+      packageJson.workspaces.nohoist.push(sourceMaps)
+    }
```

Comment:
Does this make sense all the time, or only when `workspaces` is already defined? Would creating a `workspaces` entry have other consequences for users' projects?

Reply:
We do need to add the `workspaces` parameter if it's not available, and outside of a yarn workspaces repository, will have no effect. There may well be some way to detect whether we're using yarn workspaces, though?

Reply:
if I were using this I would prefer it not to add unnecessary stuff to my package.json (if I'm not using yarn workspaces). It looks like the workspaces entry is required by definition for workspaces to be enabled. See https://classic.yarnpkg.com/lang/en/docs/workspaces/#toc-how-to-use-it - so perhaps could test with `packageJson.workspaces && packageJson.private` and not add this stuff otherwise?

Reply:
the workspaces entry needs to be present in the **root** package.json, but not the specific package. so we can't expect it to be there. but I agree that installing it when you don't need it isn't the best. will reconsider

Reply:
Updated to detect yarn workspaces before adding the nohoist configuration